parsl/executors/radical/executor.py

```diff
@@ -202,63 +227,67 @@ def start(self):
                    'resource': self.resource}
 
         if not self.resource or 'local' in self.resource:
-            # move the agent sandbox to the working dir mainly
-            # for debugging purposes. This will allow parsl
-            # to include the agent sandbox with the ci artifacts.
-            if os.environ.get("LOCAL_SANDBOX"):
-                pd_init['sandbox'] = self.run_dir
+            # set the agent dir to /temp as a fix for #3029
```

Comment:
Is this changed sandbox behaviour the only part that fixes #3029?

If so:

it's moving things into a temporary directory only when the CI tests are run (or in other situations where the `RADICAL_TEMP_SANDBOX` environment variable is set).

in non-test situations, this behaviour is otherwise unchanged, right? If so, does this PR stop that situation occurring for normal users who are running real applications, not our CI?

@AymenFJA

Reply:
I investigated this a bunch more (and have commented on what I see in other issues) to try to understand what was actually happening in the CI environment, because I like to at least have some understanding of how that environment behaves.

I can see why this is happening in the CI (due to entering a strange zone of fd number-space which radical.saga cannot deal with) and therefore why this would happen in a user workflow (when a user's workflow submitting process has opened many - around 1000 - files).

This PR avoid the misbehaving radical.saga code path by having a special non-default option to turn on a local sandbox.

But in https://github.com/radical-cybertools/radical.saga/issues/885 Andrew mentions, effectively, that radical.saga would be better replaced by psij-python - that would have the same effect of removing the misbehaving codepath, in a different way.

So I tried that out in this draft PR: https://github.com/Parsl/parsl/pull/3079 which doesn't actually avoid the misbehaving codepath, as far as I can tell.

But maybe this is a better approach longer term, if radical.saga is effectively deprecated in favour of psij-python?